internal/storage/types/ssblob/ssblob.go

```diff
@@ -0,0 +1,135 @@
+// Package ssblob provides a blob implementation for the Archivematica Storage
+// Service. Use OpenBucket to construct a *blob.Bucket.
+package ssblob
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/driver"
+	"gocloud.dev/gcerrors"
+)
+
+type NotImplemented error
+
+var errNotImpl NotImplemented
+
+type bucket struct {
+	Options
+}
+
+type Options struct {
+	Context context.Context
+	URL     string `json:"string"`
+	Base    string
+}
+
+func openBucket(opts *Options) (driver.Bucket, error) {
+	return &bucket{
+		*opts,
+	}, nil
+}
```

Comment:
This is where we would be creating the `*http.Client` and store it in the `*bucket`.

Use preferably [cleanhttp.DefaultPooledClient](https://pkg.go.dev/github.com/hashicorp/go-cleanhttp?utm_source=godoc#DefaultPooledClient). We may have to adjust some settings down the road to allow for reliable downloads, not sure.